app/src/main/java/com/example/androidtrainingproject/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,61 @@
+package com.example.androidtrainingproject.ui.home
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.example.androidtrainingproject.models.ProductResponse
+import com.example.androidtrainingproject.repository.DefaultRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class HomeViewModel @Inject constructor(
+    private val repository: DefaultRepository
+): ViewModel() {
+    var isLoading by mutableStateOf(false)
+    var fetchProductsError by mutableStateOf<Boolean?>(null)
+
+    var allProducts by mutableStateOf<List<ProductResponse>>(emptyList())
```

Comment:
Ideally those two should be `StateFlow`s in the view model but if you have the time, try out at least `mutableStateListOf`, even though it's meant to be used primarily in composables.